cookbook/upgrade/bundles.rst

```diff
@@ -0,0 +1,155 @@
+version. The easiest way to detect deprecations is to install the `PHPUnit Bridge`_
+component and then run the test suite.
+
+First, install the component as a ``dev`` dependency of your bundle:
+
+.. code-block:: bash
+
+    $ composer require --dev "symfony/phpunit-bridge"
+
+Then, run your test suite and look for the deprecation list displayed after the
+PHPUnit test report:
+
+.. code-block:: bash
+
+    $ phpunit
+
+    // ... PHPUnit output ...
+
+    Remaining deprecation notices (3)
+
+    The "pattern" option in file ... is deprecated since version 2.2 and will be
+    removed in 3.0. Use the "path" option in the route definition instead ...
+
+    Twig Function "form_enctype" is deprecated. Use "form_start" instead in ...
+
+    The Symfony\Component\Security\Core\SecurityContext class is deprecated since
+    version 2.6 and will be removed in 3.0. Use ...
+
+Fix the reported deprecations, run the test suite again and repeat the process
+until no deprecation usage is reported.
+
+Useful Resources
+~~~~~~~~~~~~~~~~
+
+There are several resources that can help you detect, understand and fix the use
+of deprecated features:
+
+* `Official Symfony Guide to Upgrade from 2.x to 3.0`_, the full list of changes
+  required to upgrade to Symfony 3.0 and grouped by component.
+* `SensioLabs DeprecationDetector`_, it runs a static code analysis against your
+  project's source code to find usages of deprecated methods, classes and
+  interfaces. It works for any PHP application, but it includes special detectors
+  for Symfony application, where it can also detect usages of deprecated services.
+* `Symfony Upgrade Fixer`_, it analyzes Symfony projects to find deprecations. In
+  addition it solves automatically some of them thanks to the growing list of
+  supported "fixers".
+
+Test your Bundle in Symfony 3
+-----------------------------
+
+Now that your bundle has removed all deprecations, it's time to test it for real
+in a Symfony 3 application. Assuming that you already have a Symfony 3 application,
+you can test the updated bundle locally without having to install it through
+Composer.
+
+If your operating system supports symbolic lings, just point the appropriate
+vendor directory to your local bundle root directory:
+
+.. code-block:: bash
+
+    $ ln -s /path/to/your/local/bundle/ vendor/you-vendor-name/your-buncle-name
+
+If your operating system doesn't support symbolic links, you'll need to copy
+your local bundle directory into the appropriate directory inside ``vendor/``.
+
+Update the Travis CI Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+In addition to running tools locally, it's recommended to set-up Travis CI service
+to run the tests of your bundle using different Symfony configurations. Use the
+following recommended configuration as the starting point of your own configuration:
+
+.. code-block:: yaml
+
+    language: php
+    sudo: false
+    php:
+        - 5.3
+        - 5.6
+        - 7.0
```

Comment:
But I want to provide a Travis config that can be copy-pasted (and tweaked/improved later on).